### DISCUSSION TITLE
produce nupkg

### DISCUSSION THREAD

**Discussion by ionutdiaconu** [ClearFoundry/ClearScript#484] [Q&A] [answered]
Hello,

I am interested in reproducing the nupkg as present in nuget.org while compiling the codebase here: https://github.com/Microsoft/ClearScript/tree/7.3.4
V8Update.cmd does not produce nupkg;
https://microsoft.github.io/ClearScript/Details/Build.html does not describe nupkg generation.

Thanks for any sugestion,
Ionut Diaconu

**Comment by ClearScriptLib** [comment]
Hi @ionutdiaconu,

You can see the package metadata, dependencies, and file structure by using the [NuGet Package Explorer](https://nuget.info/packages).

For example, [here](https://nuget.info/packages/Microsoft.ClearScript.Core) is the page for Microsoft.ClearScript.Core.

Good luck!

**Reply by ionutdiaconu** [reply]
thank you.

I am actually interested to figure out the script or command which  will produce me the following:
microsoft.clearscript.core.7.3.4.nupkg
microsoft.clearscript.v8.native.win-x64.7.3.4.nupkg

Thanks,
Ionut

**Comment by ClearScriptLib** [comment]
Hi @ionutdiaconu,

>I am actually interested to figure out the script or command which will produce me the following:
microsoft.clearscript.core.7.3.4.nupkg
microsoft.clearscript.v8.native.win-x64.7.3.4.nupkg

Here are the NuGet Package Explorer pages for the specific versions of those packages:
- [Microsoft.ClearScript.Core.7.3.4](https://nuget.info/packages/Microsoft.ClearScript.Core/7.3.4)
- [Microsoft.ClearScript.V8.Native.win-x64.7.3.4](https://nuget.info/packages/Microsoft.ClearScript.V8.Native.win-x64/7.3.4)

Cheers!

**Reply by ionutdiaconu** [reply]
thanks!

But i don't want to use the official packages.
My use-case requires me to compile clearscript myself... and finally to have nupkg packages containing in house compiled binaries.

Best Regards,
Ionut

**Comment by ClearScriptLib** [comment, marked as answer]
Hi Ionut,

>My use-case requires me to compile clearscript myself... and finally to have nupkg packages containing in house compiled binaries.

If you can build ClearScript, generating NuGet packages should be straightforward. The relevant documentation is [here](https://learn.microsoft.com/en-us/nuget/quickstart/create-and-publish-a-package-using-the-dotnet-cli), [here](https://learn.microsoft.com/en-us/nuget/quickstart/create-and-publish-a-package-using-visual-studio), and [here](https://learn.microsoft.com/en-us/nuget/quickstart/create-and-publish-a-package-using-visual-studio-net-framework). You can use the links above if you wish to use the official package internals as a guide.

Please let us know if you have difficulty building ClearScript or authoring your own NuGet package manifests.

Good luck!
